qcodes/data/data_array.py

```diff
@@ -614,12 +614,12 @@ def xarray_data_array_dictionary_to_data_array(
         preset_data = np.array(array_dictionary["data"])
     array_name = array_dictionary.get("name", array_id)
 
-    array_full_name = array_dictionary.get("long_name", array_name)
+    array_full_name = array_dictionary["attrs"].get("long_name", array_name)
     data_array = DataArray(
         name=array_name,
         full_name=array_full_name,
-        label=array_dictionary.get("label", ""),
-        unit=array_dictionary.get("unit", None),
+        label=array_dictionary["attrs"].get("long_name", ""),
+        unit=array_dictionary["attrs"].get("units", None),
```

Comment:
@peendebak could you explain, has this never worked before? if it actually did, shall the code adapt to both behvaiours, that is, try got get units/long_name from attrs AND try to get label/unit from array_dictionary as a fallback?

Reply:
@astafan8 It did not work before. The `label` and `unit`  did end up in the intermediate dictionary representation, but they did not end up in the `xarray` dataset.

The unittest in the current master has:
```
...
        for key in ["name", "unit", "label"]:
            self.assertEqual(da.name, da_transformed.name)
...
```
It is clear that the `label` and `unit` are not tested on.

I would prefer not to try to keep the old behavior intact.